app/src/main/java/com/nervousfish/nervousfish/activities/MainActivity.java

```diff
@@ -69,26 +72,28 @@ protected void onCreate(final Bundle savedInstanceState) {
         final Toolbar toolbar = (Toolbar) this.findViewById(R.id.toolbar_main);
         this.setSupportActionBar(toolbar);
 
-        if (getSupportActionBar() != null) {
-            getSupportActionBar().setDisplayShowTitleEnabled(false);
+        if (this.getSupportActionBar() != null) {
+            this.getSupportActionBar().setDisplayShowTitleEnabled(false);
         }
 
         // Fill database with demo data
         try {
-            this.contacts = this.serviceLocator.getDatabase().getAllContacts();
+            this.contacts = this.database.getAllContacts();
         } catch (final IOException e) {
             LOGGER.error("Failed to retrieve contacts from database", e);
         }
 
+        final IBluetoothHandler bluetoothHandler = this.serviceLocator.getBluetoothHandler();
         // Start Bluetooth
         try {
-            this.serviceLocator.getBluetoothHandler().start();
-        } catch (NoBluetoothException e) {
-            LOGGER.info("Bluetooth not available on device, disabling button");
+            //noinspection LawOfDemeter because we don't want to clutter the service locator
```

Comment:
Because I would have to add a method named "startBluetoothHandler" in ServiceLocator and well.. That's not what the service locator should have imo